worblehat-acceptancetests/src/test/java/de/codecentric/psd/worblehat/AcceptanceTestsIT.java

```diff
@@ -5,14 +5,5 @@
 import org.junit.runner.RunWith;
 
 @RunWith(Cucumber.class)
-@CucumberOptions(
-    strict = true,
-    stepNotifications = true,
-    plugin = {
-      "pretty",
-      "html:target/cucumber",
-      "junit:target/cucumber.xml",
-      "json:target/cucumber-report.json"
-    },
-    publish = false)
```

Comment:
Why have these options been removed from the Annotation? I've seen there's now some Kotline-File instead. How does that impact when I run the acceptance-tests from within the IDE?

Reply:
> Why have these options been removed from the Annotation? I've seen there's now some Kotline-File instead. 

I moved this to the build configuration to support build caching for cucumber outputs. I also did something like this for the test containers video. But let me start from the begining:

Gradle's build caching works based on [task inputs and outputs](https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_details). Task authors need to specify all inputs to a task that yield a specific output. For example the default inputs to the test task are the test runtime classpath (the classes under test, the test classes, as well as the jars from your dependencies). If the test task is executed Gradle will store everything that is declared as an output in the build cache. The default outputs are the HTML report as well as the XML test results. If the task is executed again with the same inputs and the outputs are missing, they will be loaded from the cache instead of reexecuting the tests.

With this cucumber configuration, the first obvious step would have been to change the configuration from `target/*` to `build/*`. However this would not make Gradle aware of these additional outputs and thus would not add them to the cache entry. For that reason I added some build configuration to inject the output locations into the tests.

> How does that impact when I run the acceptance-tests from within the IDE?

Depends on how the IDE is set up. With IntelliJ I recommend the configure 'Delegate all build actions to Gradle' in the settings. This will cause IntelliJ to start a Gradle build even if you just execute a single test and thus will make sure the configuration is injected. The same should work with Eclipse and the Buildship plugin. With other IDEs you need to make sure they delegate build actions to Gradle or need to manually add the system properties that are injected by the `CucumberOptions` CommandLineArgumentProvider. Otherwise the configuration will not be picked up.

In general you should always try to delegate build actions from the IDE to Gradle. Otherwise it's likely that you will see inconsistent results and strange errors because the IDE does not know all about how the application is build and how tests need to be executed. Having several ways to build and run application and tests is a recipe for desaster in my opinion.